src/sclient/ScanVertexIter.cpp

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "common/clients/storage/GraphStorageClient.h"
+#include "common/interface/gen-cpp2/storage_types.h"
+
+#include "nebula/sclient/ScanVertexIter.h"
+
+namespace nebula {
+
+ScanVertexIter::ScanVertexIter(const ScanVertexIter& rhs)
+    : client_(rhs.client_),
+      req_(new storage::cpp2::ScanVertexRequest(*rhs.req_)),
+      hasNext_(rhs.hasNext_),
+      nextCursor_(rhs.nextCursor_) {}
+
+ScanVertexIter::ScanVertexIter(ScanVertexIter&& rhs) noexcept
+    : client_(rhs.client_), hasNext_(rhs.hasNext_), nextCursor_(std::move(rhs.nextCursor_)) {
+    req_ = rhs.req_;
+    rhs.req_ = nullptr;
+}
+
+ScanVertexIter& ScanVertexIter::operator=(const ScanVertexIter& rhs) {
+    if (this != &rhs) {
+        client_ = rhs.client_;
+        delete req_;
+        req_ = new storage::cpp2::ScanVertexRequest(*rhs.req_);
+        hasNext_ = rhs.hasNext_;
+        nextCursor_ = rhs.nextCursor_;
+    }
+
+    return *this;
+}
+
+ScanVertexIter& ScanVertexIter::operator=(ScanVertexIter&& rhs) noexcept {
+    client_ = rhs.client_;
+    delete req_;
+    req_ = rhs.req_;
+    rhs.req_ = nullptr;
+    hasNext_ = rhs.hasNext_;
+    nextCursor_ = rhs.nextCursor_;
+
+    return *this;
+}
+
+ScanVertexIter::~ScanVertexIter() {
+    delete req_;
+}
+
+DataSet ScanVertexIter::next() {
+    if (!hasNext_) {
+        LOG(ERROR) << "hasNext() == false !";
+        return DataSet();
+    }
+    DCHECK(!!req_);
+    req_->set_cursor(std::move(nextCursor_));
```

Comment:
why use move? it will call next multi time

Reply:
because the value of nextCursor_ will change next time